docs/configuration-files.md

```diff
@@ -224,6 +226,64 @@ rules:
     severity: ERROR
 ```
 
+### `metavariable-comparison`
+
+The `metavariable-comparison` operator compares metavariables against a basic [Python comparison](https://docs.python.org/3/reference/expressions.html#comparisons)
+expression. This is useful for filtering results based on a [metavariable's](/docs/pattern-features.md#metavariables) value.
+
+**Example**
+
+The `metavariable-comparison` operator is a mapping which requires the
+`metavariable` and `comparison` keys. It can be combined with other pattern operators:
+
+```yaml
+rules:
+  - id: superuser-port
+    patterns:
+      - pattern: set_port($ARG)
+      - metavariable-comparison:
+          metavariable: '$ARG'
+          comparison: '$ARG < 1024'
+    message: "module setting superuser port"
+    languages: [python]
+    severity: ERROR
+```
+
+This will catch code like `set_port(80)` or `set_port(443)`, but not `set_port(8080)`.
```

Comment:
Could we also document whether it can compare strings to, for instance, ensure alphabetical ordering?

Reply:
Although `semgrep-core` is wired up for performing basic string comparisons I didn't include this in the Python interface. The `pattern-regex` operator can be used for string operations. I tend to split these use-cases in my mind like this: `pattern-regex` is useful for performing string operations and `pattern-comparison` is useful for performing numeric operations.

Reply:
> ... and then ensuring $FIRST < $SECOND to check for alphabetical order in a whole file.

I don't think I understand what you're trying to achieve here. You want to determine if a non-code file is in alphabetical order? Like a CSV file? This can't be done with a regex, especially because these operators only handle a single metavariable.

> Could we specify in the docs that only numeric comparisons are supported, then?

Sure thing :+1:

Reply:
I meant, in a `constants.py`, which looks like:

```py
FOO_LOGIN = "me"
FOO_PASS = "hunter2"
BAR_PASS = "hunter3"
```

I'd write

```yaml
  pattern: |
    $FIRST = ...
    $SECOND = ...
  metavariable-comparison:
    comparison: $FIRST < $SECOND
```

to ensure that `BAR_PASS` is defined first and thus the whole file is in alphabetical order.

Reply:
Reference: https://eslint.org/docs/rules/sort-vars

Reply:
Oh I see what you mean. That behavior is not currently supported. Lexicographical ordering of strings seems like a fairly niche use-case that I'd like to avoid for the v1 of this functionality in the interest of moving forward.